### ISSUE TITLE
Replace marshmallow with fastjsonschema

### ISSUE THREAD

**Issue by Luna-Klatzer** [Luna-Klatzer/openhiven.py#55]
**Summary:**
After implementing marshmallow and the release of v0.1.3.dev1, the biggest issue with marshmallow started to appear: it's slow validation. To solve that at first I thought of `toastedmarshmallow`, which sadly turned out to be not a very great solution since it's rarely updated, not really future-secure and lacks major updates and integration of the newest marshmallow version since currently it automatically installs its own fork which causes major issues trying to update or use newer versions of marshmallow.

To solve that there are multiple options, using partly 'deprecated' `toastedmarshmallow` or using a new library. Here is where `fastjsonschema` could help out a lot since it's speed and usage of classic JSON-schemas provides great functionality that can be very well used in OpenHiven.py.

**Test Code snippets:**
```python
from marshmallow import Schema, fields, INCLUDE
import time

  # normal data with one extra unexpected field
  data = {'a': 'test', 'b': 'test', 'c': 'test', 'd': 'test', 'e': 'test', 'f': 'test', 'g': 'test', 'h': 'test'}

class UserSchema(Schema):
    a = fields.Str()
    b = fields.Str()
    c = fields.Str()
    d = fields.Str()
    e = fields.Str()
    f = fields.Str()
    g = fields.Str()
SCHEME = UserSchema()

compile_dict = {
            'type': 'object',
            'properties': {
                'a': {'type': 'string'},
                'b': {'type': 'string'},
                'c': {'type': 'string'},
                'd': {'type': 'string'},
                'e': {'type': 'string'},
                'f': {'type': 'string'},
                'g': {'type': 'string'},
            }
        }
validator = fastjsonschema.compile(compile_dict)

def compare(i):
    range_ = range(i)
    
    # marshmallow
    start = time.time()
    for _ in range_:
        SCHEME.load(data, unknown=INCLUDE)
    print(f"Duration >> {float(time.time()- start)}")

    # fastjsonschema
    start = time.time()
    validator = fastjsonschema.compile(compile_dict)

    for _ in range_:
        validator(data)
    print(f"Duration >> {float(time.time()- start)}")


def compile_and_compare(i):
    range_ = range(i)
    
    # marshmallow
    start = time.time()
    class UserSchema(Schema):
        a = fields.Str()
        b = fields.Str()
        c = fields.Str()
        d = fields.Str()
        e = fields.Str()
        f = fields.Str()
        g = fields.Str()
    SCHEME = UserSchema()

    for _ in range_:
        SCHEME.load(data, unknown=INCLUDE)
    print(f"Duration >> {float(time.time() - start)}")

    # fastjsonschema
    start = time.time()
    validator = fastjsonschema.compile(compile_dict)

    for _ in range_:
        validator(data)
    print(f"Duration >> {float(time.time() - start)}")
    
def extra_compile_and_compare(i):
    range_ = range(i)

    # marshmallow
    class UserSchema(Schema):
        a = fields.Str()
        b = fields.Str()
        c = fields.Str()
        d = fields.Str()
        e = fields.Str()
        f = fields.Str()
        g = fields.Str()
    start = time.time()
    for _ in range_:
        UserSchema().load(data, unknown=INCLUDE)
    print(f"Duration >> {float(time.time() - start)}")

    # fastjsonschema
    start = time.time()
    for _ in range_:
        validator = fastjsonschema.compile(compile_dict)
        validator(data)
    print(f"Duration >> {float(time.time() - start)}")

compare(insert_num)
compile_and_compare(insert_num)
extra_compile_and_compare(insert_num)
```

**Speed results: (seconds)**

lib | type | 500 | 1000 | 2500 | 5000 | 7500 | 10000 | 25000 | 50000 | 75000 | 100000 | 150000 | 200000
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
marshmallow | compare | 0.0179975 | 0.02800035 | 0.07200003 | 0.14599824 | 0.20899773 | 0.28600049 | 0.69303107 | 1.38759804 | 2.10770941 | 3.0460391 | 4.14203477 | 5.486726522445679
fastjsonschema | compare | 0.00100374 | 0.0039978 | 0.00500035 | 0.00799942 | 0.01200223 | 0.01500034 | 0.0389986 | 0.06952572 | 0.10700107 | 0.14199781 | 0.20699239 | 0.27499890327453613
marshmallow | compile | 0.01499724 | 0.02899981 | 0.07299972 | 0.14000034 | 0.20599723 | 0.27399969 | 0.70000029 | 1.36847949 | 2.08960795 | 3.01091981 | 4.20900369 | 5.859887361526489
fastjsonschema | compile | 0.00200081 | 0.00200009 | 0.0050056 | 0.00799966 | 0.01100063 | 0.01399922 | 0.03600025 | 0.06999636 | 0.10500288 | 0.16200423 | 0.20795822 | 0.2709972858428955
marshmallow | extra | 0.06901765 | 0.13400006 | 0.33099413 | 0.65099978 | 0.99818206 | 1.36506391 | 3.41458797 | 6.57011795 | 9.99148583 | 14.0237195 | 20.0939629 | 26.759239435195923
fastjsonschema | extra | 0.37799859 | 0.73599792 | 1.86587453 | 3.66753459 | 5.34444523 | 7.47883129 | 18.5129743 | 38.2523735 | 55.2930806 | 79.3817549 | 108.485124 | 147.89626383781433

![image](https://user-images.githubusercontent.com/61756091/107307758-95208e00-6a87-11eb-8739-5de72c845a27.png)

**Conclusion:**
Based on the results, it's evident that marshmallow is fast with a few objects but quickly gets slower and slower, the more input you give. While fastjsonschema has the same effect, it's never surpassing the 1-second mark, making it very well suited for a lot of data-validation that uses the same schema. If every object is different and needs to be always newly compiled, fastjsonschema clearly is losing to marshmallow since the compilation skyrockets after only 10000 objects

**Comment by Nicolas-Klatzer** [comment]
Will be implemented with the Rewrite (#54), defaulting for now to marshmallow until the JSON cache was implemented #53

**Comment by Luna-Klatzer** [comment]
Closed with #65
